### Import Libraries

In [1]:
import pandas as pd
from scipy import stats

### Import Data

In [2]:
df=pd.read_csv("Costomer+OrderForm.csv")
df.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

### EDA

In [3]:
df1=df.apply(pd.Series.value_counts)

In [4]:
observed=df1.to_numpy()
observed

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]], dtype=int64)

### Hypothesis Test

In [5]:
stat,p,dof,expected=stats.chi2_contingency(observed)

In [6]:
print(stat,p,dof,expected)

3.858960685820355 0.2771020991233135 3 [[271.75 271.75 271.75 271.75]
 [ 28.25  28.25  28.25  28.25]]


**HYPOTHESIS**

Ho : defective % does not vary by centre

Ha : defective % varies by centre

Observed = [[271, 267, 269, 280]

[ 29,  33,  31,  20]]

**SOLUTION**

Expected = [[271.75 271.75 271.75 271.75]

[ 28.25  28.25  28.25  28.25]]

p value = 0.27, alpha = 0.05

p value is greater than alpha. Hence Ho is failed to reject.

**CONCLUSION**

defective % does not vary by centre

### End !!!